### PR TITLE
Change output format for price feed check

### PR DESCRIPTION
**Pull request by nivcertora**
https://certora.atlassian.net/browse/CERT-7874
https://certora.atlassian.net/browse/CERT-7875

### **PR Summary**

**Refactor PriceFeedData and Enhance PriceFeedCheck Messaging**

- **Remove Address Fields:**
  - Eliminated `address` and `proxy_address` from the `__str__` method in `PriceFeedData` to streamline its string representation.

- **Enhance Messaging in PriceFeedCheck:**
  - Updated `PriceFeedCheck` to include `proxy_address` and `origin_address` in success messages when they are available and differ from the primary address.
  - Provides more detailed information about the price feed sources, improving clarity and debugging capabilities.

image

### CODE REVIEW

**Review by yoav-el-certora — DISMISSED** [review]
The output is not as required by ticket.

For price feed checks we expect:
```
Found <address> on <price feed provider>
Name: <price feed name>
Decimals: <decimals>

<if this is an implementation address> <in red> this is an implementation contract with a proxy address <proxy address> </in red>
```

For coin gecko we expect:
```
Found <address> on <token validator provider>
Name: <token name>
Symbol: <token symbol>
Decimals: <decimals>
```

**Comment by nivcertora** [comment]
> The output is not as required by ticket.
> 
> For price feed checks we expect:
> 
> ```
> Found <address> on <price feed provider>
> Name: <price feed name>
> Decimals: <decimals>
> 
> <if this is an implementation address> <in red> this is an implementation contract with a proxy address <proxy address> </in red>
> ```
> 
> For coin gecko we expect:
> 
> ```
> Found <address> on <token validator provider>
> Name: <token name>
> Symbol: <token symbol>
> Decimals: <decimals>
> ```

The output is literally the same, you can check the output image.

File: Quorum/checks/price_feed.py
Comment (yoav-el-certora) [inline Quorum/checks/price_feed.py:86]: Only this sentence should be printed in red

File: Quorum/apis/price_feeds/price_feed_utils.py
Comment (yoav-el-certora) [inline Quorum/apis/price_feeds/price_feed_utils.py:40]: Replace "Pair" with "Symbol"

File: Quorum/checks/price_feed.py
Comment (nivcertora) [reply Quorum/checks/price_feed.py:87]: <if this is an implementation address> <in red> this is an implementation contract with a proxy address <proxy address> </in red>

File: Quorum/checks/price_feed.py
Comment (nivcertora) [reply Quorum/checks/price_feed.py:86]: It's not possible to do. and better to put the entire message to not miss it.

File: Quorum/checks/price_feed.py
Comment (yoav-el-certora) [reply Quorum/checks/price_feed.py:87]: This line - message += f"Origin Address: {price_feed.address}\n"

File: Quorum/apis/price_feeds/price_feed_utils.py
Comment (nivcertora) [reply Quorum/apis/price_feeds/price_feed_utils.py]: Done

File: Quorum/checks/price_feed.py
Comment (yoav-el-certora) [reply Quorum/checks/price_feed.py:86]: But everything in red makes you think this is a failure

Comment (nivcertora) [reply Quorum/checks/price_feed.py:86]: It is a failure according to the ticket.

Comment (nivcertora) [reply Quorum/checks/price_feed.py:86]: If the contract uses the proxy address instead of the original they want it in red

File: Quorum/checks/price_feed.py
Comment (nivcertora) [reply Quorum/checks/price_feed.py:87]: It was but they made a mistake, they don't need the proxy address in the sentence as it is already printed. they want the original.

File: Quorum/checks/price_feed.py
Comment (yoav-el-certora) [reply Quorum/checks/price_feed.py:86]: From ticket description, they want only the last line.
Please check this with Michael.

Comment (nivcertora) [reply Quorum/checks/price_feed.py:86]: ill merge it and if have a issue ill talk to him.

Comment (yoav-el-certora) [reply Quorum/checks/price_feed.py:86]: Check with @MichaelMorami before merge.

**Pull request merged**
